src/async-tokenize/tests/async-tokenize.test.ts

```diff
@@ -0,0 +1,332 @@
+import { fireEvent, waitFor } from '@testing-library/dom'
+import {
+  formElementsMock,
+  formValuesMock,
+  handleFormMock,
+} from '../../../tests/mocks/elements-values-mocks'
+import { Malga } from '../../common/malga/malga'
+import { AsyncTokenize } from '../async-tokenize'
+import * as utilsValues from '../../common/utils/form-values/form-values'
+import * as utilsElements from '../../common/utils/form-elements/form-elements'
+
+const MalgaConfigurations = {
+  apiKey: '17a64c8f-a387-4682-bdd8-d280493715e0',
+  clientId: 'd1d2b51a-0446-432a-b055-034518c2660e',
+  options: {
+    sandbox: true,
+  },
+}
+const onSubmit = vi.fn()
+
+vi.mock('../../common/malga/malga', async (importOriginal) => {
+  const Malga =
+    await importOriginal<typeof import('../../common/malga/malga')>()
+  return {
+    ...Malga,
+    tokenization: vi.fn(),
+  }
+})
+
+function Form(props: any) {
```

Comment:
Tenho dois pontos sobre essa função:

1. Essa função provavelmente poderia ter um nome mais descritivo! Para que seja mais fácil entender o que ela faz ao longo do código quando ela for chamada. Pelo que eu entendi, ela monta um form com os dados da Taylor Swift, então acho que ela poderia se chamar `generateForm()` ou `settingUpForm()`, ou algo parecido!

2. Ela não precisa dessa props, porque em todas as chamadas da função ela recebe o mesmo parâmetro `onSubmit` que por si só também poderia estar numa constante! Assim, reduzimos caracteres e aumentamos legibilidade!

Reply:
fiquei em dúvida entre generateForm e renderForm, mas optei pelo primeiro